contracts/serverless-v2/Relay.sol

```diff
@@ -882,18 +882,21 @@ contract Relay is
         uint8 _errorCode,
         uint256 _signTimestamp
     ) internal {
-        Job memory job = jobSubscriptions[_jobSubsId].job;
-        if (job.jobOwner == address(0)) 
+        JobSubscription memory jobSubs = jobSubscriptions[_jobSubsId];
+        if (jobSubs.job.jobOwner == address(0)) 
             revert RelayInvalidJobSubscription();
 
-        // check time case
-        // if (block.timestamp > job.startTime + OVERALL_TIMEOUT) revert RelayOverallTimeoutOver();
+        // getting the virtual start time of the job subscription current run
+        uint256 jobStartTime = jobSubs.job.startTime + (jobSubs.currentRuns * jobSubs.periodicGap);
+        if(block.timestamp > jobStartTime + OVERALL_TIMEOUT)
+            revert RelayOverallTimeoutOver();
 
+        // adjusting currentRuns to last 127 bits of jobSubsId to prevent txn replay
+        uint256 jobId = _jobSubsId | (jobSubs.currentRuns + 1);
```

Comment:
We are not expecting number of runs to be always incremented by 1. It can be like 8, 10, 11.  9 is missing here. But this contract still has to allow number 10 to be submitted. So, the check you need is if instance count in the job ID provided by the gateway is greater than currentRuns. Also, once check succeeds, set the currentRuns to instance count provided by the gateway as 127 LSB of Job_ID.